pkg/action/lint.go

```diff
@@ -59,15 +59,22 @@ func (l *Lint) Run(paths []string, vals map[string]interface{}) *LintResult {
 
 	result := &LintResult{}
 	for _, path := range paths {
-		if linter, err := lintChart(path, vals, l.Namespace, l.Strict); err != nil {
+		linter, err := lintChart(path, vals, l.Namespace, l.Strict)
+		if err != nil {
 			if err == errLintNoChart {
 				result.Errors = append(result.Errors, err)
 			}
+			if linter.HighestSeverity >= lowestTolerance {
```

Comment:
These changes feel out of scope for this PR because it doesn't seem to be related to the values map mutation. Please let me know if I am wrong

Reply:
None of the `Messages` which are errors are being appended to `Result.Errors` which shouldn't be the case.

Reply:
While writing the test, I noticed that there was <nil> getting appended to Result.Errors. This was creating an issue because, Result.Errors was supposed tohave a length of zero, instead it had a length of 1, hence failing my test even on the right code.

When running lintChart on a chart which passes the lint, it would return <nil> for err which is expected. But this lint was being appended to result.Errors which isn't the expected behaviour. Do correct me if I'm wrong.

Reply:
OK, that make sense then. So this was a different bug you found along the way and fixed also

Reply:
@waveywaves could you write a unit test to make sure we can validate that in the future?

Reply:
@thomastaylor312 I have updated the PR, can you check and let me know.